src/v2/view-builder/views/IdentifierView.js

```diff
@@ -1,34 +1,30 @@
-import { View, loc, createButton } from 'okta';
+import { loc } from 'okta';
 import BaseView from '../internals/BaseView';
 import BaseForm from '../internals/BaseForm';
 import BaseFooter from '../internals/BaseFooter';
+import signInWithIdps from './signin/SignInWithIdps';
+import signInWithDeviceOption from './signin/SignInWithDeviceOption';
 
 const Body = BaseForm.extend({
 
   title: loc('primaryauth.title'),
   save: loc('oform.next', 'login'),
   initialize () {
     BaseForm.prototype.initialize.apply(this, arguments);
-    if (this.options.appState.hasRemediationForm('launch-authenticator')) {
-      this.add(View.extend({
-        className: 'sign-in-with-device-option',
-        template: `
-          <div class="okta-verify-container"></div>
-          <div class="separation-line"><span>OR</span></div>
-        `,
-        initialize () {
-          const appState = this.options.appState;
-          this.add(createButton({
-            className: 'button-secondary',
-            title: 'Sign in using Okta Verify',
-            click () { 
-              appState.trigger('invokeAction', 'launch-authenticator');
-            }
-          }), '.okta-verify-container');
-        }
-      }), '.o-form-fieldset-container', false, true);
-    }
-  }
+    this.listenTo(this.options.appState, 'formRendered', () => {
+      if (this.options.appState.hasRemediationForm('launch-authenticator')) {
+        this.add(signInWithDeviceOption, '.o-form-fieldset-container', false, true);
+      }
+      //add idps
+      if (this.options.appState.get('idps')) {
```

Comment:
- maybe shall do `if (appState.hasRemediationForm('redirect')` to align with another case.
- `hasRemediationForm` doesn't appropriate now and could rename to `hasRemediationObject`?